x-pack/plugins/security_solution/common/api/endpoint/metadata/list_metadata_route.ts

```diff
@@ -16,6 +16,21 @@ export const GetMetadataListRequestSchema = {
       page: schema.number({ defaultValue: ENDPOINT_DEFAULT_PAGE, min: 0 }),
       pageSize: schema.number({ defaultValue: ENDPOINT_DEFAULT_PAGE_SIZE, min: 1, max: 10000 }),
       kuery: schema.maybe(schema.string()),
+      sortField: schema.maybe(
+        schema.oneOf([
+          // allowed fields for sorting - these are the column fields in the EndpointList table, based on the
+          // returned `HostInfoInterface` data type, and not on the internal data structure
+          schema.literal('metadata.host.hostname'),
+          schema.literal('host_status'),
+          schema.literal('metadata.Endpoint.policy.applied.name'),
+          schema.literal('metadata.Endpoint.policy.applied.status'),
+          schema.literal('metadata.host.os.name'),
+          schema.literal('metadata.host.ip'),
+          schema.literal('metadata.agent.version'),
```

Comment:
nit: maybe save these literals into an enum and use it here. Would also make it easy to use that in the tests that verify the sort fields.

Reply:
done! thanks for the idea 🙌 
70bf06b1bfceece78c9db3de57f405bc05700705